### PR TITLE
Update app-architecture.rst

### PR DESCRIPTION
**Pull request by carrielaben-da**
Second-iteration changes to the App Architecture page, addressing User Management and the Alias contract.

[CHANGELOG_BEGIN]
[CHANGELOG_END]

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/getting-started/app-architecture.rst [hunk 1]
```diff
@@ -53,20 +53,20 @@ In this case all users that a particular user is following are able to see the u
...
+ - Lastly we have user aliases. These are not a builtin concept, they are defined by an *Alias template* (discussed below) within the specific model used in this guide. Aliases serve as a way to address parties on all nodes via a human readable name.
 
-The social network user discussed in this guide is really a combination of all three of these concepts. Alice, Bob, and Charlie are all aliases that correspond to a single user id and party id each. 
+The social network users discussed in this guide are really a combination of all three of these concepts. Alice, Bob, and Charlie are all aliases that correspond to a single test user and a single party id each. As part of the installation process the ``Setup.daml`` file creates the three test users and their aliases, along with a public party to manage further user ids and aliases.
```
Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:59]: This is not true. `Setup.daml` does not create the aliases. They are created in the frontend when you log in. This is also the first time we mention the public party I believe. I think we should explain what that party is needed for.

File: docs/source/getting-started/app-architecture.rst [hunk 2]
```diff
@@ -53,20 +53,20 @@ In this case all users that a particular user is following are able to see the u
...
+ - Lastly we have user aliases. These are not a builtin concept, they are defined by an *Alias template* (discussed below) within the specific model used in this guide. Aliases serve as a way to address parties on all nodes via a human readable name.
 
-The social network user discussed in this guide is really a combination of all three of these concepts. Alice, Bob, and Charlie are all aliases that correspond to a single user id and party id each. 
+The social network users discussed in this guide are really a combination of all three of these concepts. Alice, Bob, and Charlie are all aliases that correspond to a single test user and a single party id each. As part of the installation process the ``Setup.daml`` file creates the initial three test users and a public party. This public party is for user management. Here, for instance, it supports creating additional users.
```
Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:59]: The public party is not for user management. It is to make aliases visible to all users. I think this deserves an explanation of observers and why we’re using a public party instead.

Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:59]: I don’t understand what you mean by `creating additional users`.

Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:59]: I think by installation you mean `daml start` (or at least that’s what we should say). Specifically daml start runs the script that you specify via `init-script` in your daml.yaml. I think this is also the first place we reference script so probably worth at least linking to that.

File: docs/source/getting-started/app-architecture.rst [hunk 3]
```diff
@@ -53,20 +53,20 @@ In this case all users that a particular user is following are able to see the u
...
- - On each participant node you can create users with human-readable user ids. Each user can be associated with a party allocated on that participant node, and refers to that party only on that node. Users are a purely local concept, meaning you can never address a user on another node by user id, and you never work with users in your Daml code; party ids are always used for these purposes. Users are also a builtin concept.
- - Lastly we have user aliases. These are not a builtin concept, they are defined by an Alias template within the specific model used in this guide. Aliases serve as a way to address parties on all nodes via a human readable name.
+ - On each participant node you can create users with human-readable user ids. Each user can be associated with one or more parties (up to 100) allocated on that participant node, and refers to that party only on that node. Users are a purely local concept, meaning you can never address a user on another node by user id, and you never work with users in your Daml code; party ids are always used for these purposes. Users are also a builtin concept.
```
Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:56]: The limit is not 100. I think it’s 10k claims but I’m struggling to find the docs.

File: docs/source/getting-started/app-architecture.rst [hunk 4]
```diff
@@ -86,6 +86,13 @@ That is what the ``Follow`` choice does: after checking some preconditions, it a
 
 More detailed information on choices can be found in :doc:`our docs </daml/reference/choices>`.
 
+Finally, the ``User.daml`` file contains the Alias template that manages the link between user ids and aliases:
```
Comment (cocreature) [inline docs/source/getting-started/app-architecture.rst:89]: I think we should tie this back to the public party and explain that we set it as observer.

File: docs/source/getting-started/app-architecture.rst (hunk 2 above)
Comment (carrielaben-da) [reply docs/source/getting-started/app-architecture.rst:59]: When you say "link" do you mean conceptually or an actual hotlink to the script? Or a literal include like we did with User.daml?

File: docs/source/getting-started/app-architecture.rst (hunk 4 above)
Comment (carrielaben-da) [reply docs/source/getting-started/app-architecture.rst:89]: Something like this? "Here the public party created by the ``Setup.daml`` file is made an observer to the contract linking the user and alias. This shares the alias with everyone."

**Review by cocreature — APPROVED** [review]

**Pull request merged**